src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -4,27 +4,46 @@
 import me.karakelley.http.Request;
 import me.karakelley.http.Response;
 import me.karakelley.http.Status;
+import me.karakelley.http.utility.HtmlGenerator;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class StaticFilesController implements Controller {
+
   private final ArrayList availableActions = new ArrayList<>(Collections.singleton("GET"));
 
   public Response respond(Request request) {
+    String requestedResource = request.getPath();
+
     Response response = new Response();
     if (request.validRequestLine() && availableActions.contains(request.getMethod())) {
-      return displayFilesResponse(response);
+      return serveDirectoryResponse(response, requestedResource);
     } else {
       return new InvalidRequestController().respond(request);
```

Comment:
This might not be the best time to address this, but I don't think this `StaticFilesController` class should have a dependency on another controller—aren't they basically "peers"? It'd be nice if you could guarantee that if you enter this class, only one thing can happen (i.e. there is _only_ a "happy path").

If you want to hold on that and do it in a separate PR I'm OK with that.

Reply:
Yeah definitely, I think if I can abstract the request validation I can remove that dependency. I am going to add it to my list.